code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+	cut_overlays()
+	if(!small_sprite.small)
+		return
+	if(stat == DEAD)
+		var/mutable_appearance/overlay = mutable_appearance(icon, "overlay_dead")
+		overlay.appearance_flags = RESET_COLOR
+		add_overlay(overlay)
+		return
+	if(!using_special)
+		var/mutable_appearance/overlay = mutable_appearance(icon, "overlay_base")
+		overlay.appearance_flags = RESET_COLOR
+		add_overlay(overlay)
+		return
+	if(using_special)
+		var/mutable_appearance/overlay = mutable_appearance(icon, "overlay_gust")
+		overlay.appearance_flags = RESET_COLOR
+		add_overlay(overlay)
+
+/**
+  * Determines a line of turfs from sources's position to the target with length range.
+  *
+  * Determines a line of turfs from the source's position to the target with length range.
+  * The line will extend on past the target if the range is large enough, and not reach the target if range is small enough.
+  * Arguments:
+  * * offset - whether or not to aim slightly to the left or right of the target
+  * * range - how many turfs should we go out for
+  * * atom/at - The target
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/line_target(offset, range, atom/at = target)
+	if(!at)
+		return
+	var/angle = ATAN2(at.x - src.x, at.y - src.y) + offset
+	var/turf/T = get_turf(src)
+	for(var/i in 1 to range)
+		var/turf/check = locate(src.x + cos(angle) * i, src.y + sin(angle) * i, src.z)
+		if(!check)
+			break
+		T = check
+	return (getline(src, T) - get_turf(src))
+
+/**
+  * Spawns fire at each position in a line from the source to the target.
+  *
+  * Spawns fire at each position in a line from the source to the target.
+  * Stops if it comes into contact with a solid wall, a window, or a door.
+  * Delays the spawning of each fire by 1.5 deciseconds.
+  * Arguments:
+  * * atom/at - The target
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/fire_stream(var/atom/at = target)
+	playsound(get_turf(src),'sound/magic/fireball.ogg', 200, TRUE)
+	var/range = 20
+	var/list/turfs = list()
+	turfs = line_target(0, range, at)
+	var/delayFire = -1.0
+	for(var/turf/T in turfs)
+		if(istype(T, /turf/closed))
+			return
+		for(var/obj/structure/window/W in T.contents)
+			return
+		for(var/obj/machinery/door/D in T.contents)
+			if(D.density)
+				return
+		delayFire += 1.5
+		addtimer(CALLBACK(src, .proc/dragon_fire_line, T), delayFire)
+
+/**
+  * What occurs on each tile to actually create the fire.
+  *
+  * Creates a fire on the given turf.
+  * It creates a hotspot on the given turf, damages any living mob with 30 burn damage, and damages mechs by 50.
+  * It can only hit any given target once.
+  * Arguments:
+  * * turf/T - The turf to trigger the effects on.
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/dragon_fire_line(turf/T)
+	var/list/hit_list = list()
+	hit_list += src
+	new /obj/effect/hotspot(T)
+	T.hotspot_expose(700,50,1)
+	for(var/mob/living/L in T.contents)
+		if(L in hit_list)
+			continue
+		hit_list += L
+		L.adjustFireLoss(30)
```

Comment:
No armour check?

Reply:
It never had this before in the old space dragon. I think armor is handled automatically? I can't find anywhere armor is used really.